components/FormToCreateShoppingItem.js

```diff
@@ -0,0 +1,108 @@
+import styled from "styled-components";
+
+export default function FormToCreateShoppingItem({ onAddItem, categories }) {
+  console.log("Categories: ", categories);
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const newItem = Object.fromEntries(formData);
+    newItem.quantity = Number(newItem.quantity);
+    newItem.imageUrl = "placeholder_1.png";
+
+    onAddItem(newItem);
+    event.target.reset();
+  }
+
+  return (
+    <article>
+      <container>
+        <h1>Add an item to the list:</h1>
+        <form onSubmit={handleSubmit} data-js="form">
+          <StyledFieldset>
+            <StyledLabel>
+              new shopping item*:
+              <StyledInput
+                name="name"
+                type="text"
+                required
+                placeholder="e.g., salt"
+              />
+            </StyledLabel>
+            <StyledLabel>
+              number*:
+              <StyledInput name="quantity" type="number" required />
+            </StyledLabel>
+            <StyledLabel>
+              category*:
+              <select
+                key="category"
+                name="category"
+                data-js="category"
+                required
+              >
+                <option value="">please select a category</option>
+                {categories.map((category) => (
+                  <option key={category.name} value={category.name}>
```

Comment:
In this case, I will stick to the key category.name as the categories themselves should be unique. Nanoid is used below for new items added to the list; in the case of the categories, there is so far no way to add new categories. Thanks for pointing it out though. It may become important later.